test/suites/consolidation/suite_test.go

```diff
@@ -63,6 +65,222 @@ var _ = AfterEach(func() { env.Cleanup() })
 var _ = AfterEach(func() { env.AfterEach() })
 
 var _ = Describe("Consolidation", func() {
+	Context("Budgets", func() {
+		var nodePool *corev1beta1.NodePool
+		var dep *appsv1.Deployment
+		var selector labels.Selector
+		var numPods int32
+		BeforeEach(func() {
+			nodePool = env.DefaultNodePool(nodeClass)
+			nodePool.Spec.Disruption.ConsolidateAfter = nil
+
+			numPods = 5
+			dep = test.Deployment(test.DeploymentOptions{
+				Replicas: numPods,
+				PodOptions: test.PodOptions{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: map[string]string{"app": "regular-app"},
+					},
+					ResourceRequirements: v1.ResourceRequirements{
+						Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("1")},
+					},
+				},
+			})
+			selector = labels.SelectorFromSet(dep.Spec.Selector.MatchLabels)
+		})
+		It("should respect budgets for empty delete consolidation", func() {
+			nodePool.Spec.Disruption.Budgets = []corev1beta1.Budget{
+				{
+					Nodes: "40%",
+				},
+			}
+
+			// Hostname anti-affinity to require one pod on each node
+			dep.Spec.Template.Spec.Affinity = &v1.Affinity{
+				PodAntiAffinity: &v1.PodAntiAffinity{
+					RequiredDuringSchedulingIgnoredDuringExecution: []v1.PodAffinityTerm{
+						{
+							LabelSelector: dep.Spec.Selector,
+							TopologyKey:   v1.LabelHostname,
+						},
+					},
+				},
+			}
+			env.ExpectCreated(nodeClass, nodePool, dep)
+
+			env.EventuallyExpectCreatedNodeClaimCount("==", 5)
+			nodes := env.EventuallyExpectCreatedNodeCount("==", 5)
+			env.EventuallyExpectHealthyPodCount(selector, int(numPods))
+
+			By("adding finalizers to the nodes to prevent termination")
+			for _, node := range nodes {
+				Expect(env.Client.Get(env.Context, client.ObjectKeyFromObject(node), node)).To(Succeed())
+				node.Finalizers = append(node.Finalizers, common.TestingFinalizer)
+				env.ExpectUpdated(node)
+			}
+
+			dep.Spec.Replicas = lo.ToPtr[int32](1)
+			By("making the nodes empty")
+			// Update the deployment to only contain 1 replica.
+			env.ExpectUpdated(dep)
+
+			// Ensure that we get two nodes tainted, and they have overlap during the drift
+			nodes = env.EventuallyExpectTaintedNodeCount("==", 2)
+			env.ConsistentlyExpectTaintedNodeCount("==", 2, "5s")
```

Comment:
nit: you should set nodes = second one since it's technically going to be more updated, and also already returns the list of nodes.

Reply:
> you should set nodes = second one since it's technically going to be more updated, and also already returns the list of nodes

Sorry, not getting what this comment means here.